packages/docusaurus-theme-common/src/utils/IntlUtils.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+
+export function useCalendar(): string {
+  const {
+    i18n: {currentLocale, localeConfigs},
+  } = useDocusaurusContext();
+  return localeConfigs[currentLocale]!.calendar;
+}
+
+export function useDateTimeFormat(
+  options: Intl.DateTimeFormatOptions = {},
+): Intl.DateTimeFormat {
+  const {
+    i18n: {currentLocale},
+  } = useDocusaurusContext();
+  const calendar = useCalendar();
+  return new Intl.DateTimeFormat(currentLocale, {
+    calendar,
+    ...options,
+  });
+}
```

Comment:
why are those changes appearing in this PR's diff?

Reply:
Merged main into this PR to resolve `formattedDate` in snapshot, reverted the commit for clearer review of this PR